src/jacobians.jl

```diff
@@ -192,7 +192,7 @@ function finite_difference_jacobian(
             return dx
         end
 
-        if jac_prototype isa Nothing
+        if jac_prototype isa Nothing && sparsity isa Nothing
```

Comment:
is there a specific file in OrdinaryDiffEq.jl that we should include as a downstream test here?

Reply:
https://github.com/SciML/OrdinaryDiffEq.jl/blob/master/test/interface/sparsediff_tests.jl probably this?

Reply:
yes that's a good one. Can you setup an environment which does a downstream test to that? That should hopefully catch any outrageous issues.